crates/polars-plan/src/logical_plan/aexpr/schema.rs

```diff
@@ -111,12 +111,22 @@ impl AExpr {
                         let mut field =
                             arena.get(*expr).to_field(schema, Context::Default, arena)?;
                         float_type(&mut field);
+                        if field.dtype == DataType::Date {
+                            field.coerce(DataType::Datetime(TimeUnit::Milliseconds, None));
+                        } else {
+                            float_type(&mut field);
+                        }
                         Ok(field)
                     },
                     Mean(expr) => {
                         let mut field =
                             arena.get(*expr).to_field(schema, Context::Default, arena)?;
                         float_type(&mut field);
```

Comment:
Do you mean the `float_type(&mut field);`? I think because the mean of integers still need the scheme to map to a float, is that not correct? Edit: ooh, I see it right above. Looks like I forgot to delete two lines in here....thanks for the catch @reswqa.